plugins/modules/foreman_scc_product.py

```diff
@@ -0,0 +1,118 @@
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+---
+module: foreman_scc_product
+short_description: Manage Foreman SccProduct
+description:
+  - "Manage Foreman SccProduct Entities"
+  - "This module requires the foreman_scc_manager plugin set up in the server"
+  - "See: U(https://github.com/ATIX-AG/foreman_scc_manager)"
+author:
+  - "Manisha Singhal (@manisha15) ATIX AG"
+options:
+  name:
+    description: Name of the scc_product
+    required: true
+    type: str
+  friendly_name:
+    description: Full name of the scc_product
+    required: false
+    type: str
+  organization:
+    description: Name of related organization
+    required: false
+    type: str
+  scc_account:
+    description: Name of the scc_account associated with scc_product
+    required: true
+    type: str
+  state:
+    description: State of the scc_product
+    default: subscribe
+    choices:
+      - subscribe
+    type: str
+extends_documentation_fragment: foreman
+'''
+
+EXAMPLES = '''
+- name: "Subscribe to scc_product"
+  foreman_scc_product:
+    name: "Product1"
+    scc_account_name: "Test1"
+    state: subscribe
+'''
+
+RETURN = ''' # '''
+
+
+from ansible.module_utils.foreman_helper import KatelloEntityAnsibleModule
+
+
+def main():
+    module = KatelloEntityAnsibleModule(
+        entity_spec=dict(
+            name=dict(required=True),
+            friendly_name=dict(),
+            scc_account=dict(required=True),
+            state=dict(default='subscribe', choices=['subscribe']),
+        ),
+    )
+
+    entity_dict = module.clean_params()
+
+    module.connect()
+
+    entity_dict['organization'] = module.find_resource_by_name('organizations', entity_dict['organization'], thin=True)
+
+    scope = {'organization_id': entity_dict['organization']['id']}
+
+    entity_dict['scc_account'] = module.find_resource_by_name('scc_accounts', name=entity_dict['scc_account'], params=scope, thin=True)
+
+    scope = {'scc_account_id': entity_dict['scc_account']['id']}
+
+    # Try to find the SccProduct to work on
+    # name is however not unique, but friendly_name is
+    entity = None
+    # If we have a friendly_name, search for it
+    if 'friendly_name' in entity_dict and entity_dict['friendly_name'] != '':
+        search_string = 'friendly_name="{0}"'.format(entity_dict['friendly_name'])
+        entity = module.find_resource('scc_products', search_string, params=scope, failsafe=True)
+
+    if module.state == 'subscribe':
+        result = module.foremanapi.resource('scc_products').call('subscribe', {'id': entity['id'], 'scc_account_id': scope['scc_account_id']})
```

Comment:
```suggestion
        params = {'id': entity['id']}
        params.update(scope)
        result = module.foremanapi.resource('scc_products').call('subscribe', params)
```

also, any reason not to use `resource_action` here?